conans/test/unittests/client/generators/visual_studio_multi_test.py

```diff
@@ -19,6 +20,7 @@
 
 @attr('visual_studio')
 @pytest.mark.tool_visual_studio
+@unittest.skipUnless(platform.system() == "Windows", "Requires Windows")
```

Comment:
The truth is that this test is a unittest, that shouldn't rely on the current platform. Just taking some hardcoded settings and outputting an xml file, should be completely platform independent (and it was passing in the past, right?) 

What is the difference now? What is failing?

Reply:
It's because we have pytest configured to skip tests that try to run when a marked tool is missing, as it was marked for visual studio with pytest it was skipped for pytest but ran with nose:
https://github.com/conan-io/conan/blob/639450053a0359af5a757b3617a82a2ece2f7cf1/conans/test/conftest.py#L53-L56

Reply:
Oh, I see, then isn't it possible that the solution would be remove the ``@pytest.mark.tool_visual_studio``? As the test doesn't need Visual Studio installed at all. Do we really want that this unittest only run on Windows?

Reply:
This is exactly my main concern with adding marks to tests and use them to skip tests when the mark is not used... that a wrong mark will prevent the test from running. Every test should run unless something is added to the command line to skip it, not the other way around.